File: warrior/src/main/java/com/dumbdogdiner/warrior/managers/GameBarManager.java
package com.dumbdogdiner.warrior.managers;

import com.dumbdogdiner.warrior.api.managers.WarriorGameBarManager;
import com.dumbdogdiner.warrior.user.User;
import com.dumbdogdiner.warrior.api.builders.GameBossBar;
import org.bukkit.boss.BarColor;

import java.util.TreeMap;

public class GameBarManager implements WarriorGameBarManager {

    private final TreeMap<User, GameBossBar> bossBars = new TreeMap<>();

    public void addPlayer(User user, GameBossBar bossBar) {
        if(bossBars.get(user) != null)
            bossBars.remove(user);

        bossBars.put(user, bossBar);
        bossBar.showPlayer(user.getBukkitPlayer());
    }

    public void removePlayer(User user) {
        if(bossBars.get(user) != null) {
            GameBossBar bossBar = bossBars.get(user);
            bossBar.removePlayer(user.getBukkitPlayer());
            bossBars.remove(user);
        }
    }

    public GameBossBar getBossBar(User user) {
        return bossBars.get(user);
    }

    public void updateTitle(User user, String title) {
        if(bossBars.get(user) != null) {
            GameBossBar bossBar = bossBars.get(user);
            bossBar.setTitle(title);
        }
    }

    public void updateColor(User user, BarColor color) {
        if(bossBars.get(user) != null) {
            GameBossBar bossBar = bossBars.get(user);
            bossBar.setColor(color);
        }
    }

    public void updateProgress(User user, double val) {
        if(bossBars.get(user) != null) {
            GameBossBar bossBar = bossBars.get(user);
            bossBar.setProgress(val);
        }
    }

}

File: warrior/src/main/kotlin/com/dumbdogdiner/warrior/managers/GameBarManager.kt
package com.dumbdogdiner.warrior.managers

import com.dumbdogdiner.warrior.api.managers.WarriorGameBarManager
import java.util.TreeMap
import com.dumbdogdiner.warrior.api.builders.GameBossBar
import com.dumbdogdiner.warrior.api.user.WarriorUser
import com.dumbdogdiner.warrior.user.User
import org.bukkit.boss.BarColor

class GameBarManager : WarriorGameBarManager {

    private val bossBars = TreeMap<WarriorUser, GameBossBar?>()

    override fun addPlayer(user: WarriorUser, bossBar: GameBossBar) {
        if (bossBars[user] != null) bossBars.remove(user)
        bossBars[user] = bossBar
        bossBar.showPlayer(user.bukkitPlayer)
    }

    override fun removePlayer(user: WarriorUser) {
        if (bossBars[user] != null) {
            val bossBar = bossBars[user]
            bossBar!!.removePlayer(user.bukkitPlayer)
            bossBars.remove(user)
        }
    }

    override fun getBossBar(user: WarriorUser): GameBossBar? {
        return bossBars[user]
    }

    override fun updateTitle(user: WarriorUser, title: String?) {
        if (bossBars[user] != null) {
            val bossBar = bossBars[user]
            bossBar!!.setTitle(title)
        }
    }

    override fun updateColor(user: WarriorUser, color: BarColor?) {
        if (bossBars[user] != null) {
            val bossBar = bossBars[user]
            bossBar!!.setColor(color)
        }
    }

    override fun updateProgress(user: WarriorUser, double: Double) {
        if (bossBars[user] != null) {
            val bossBar = bossBars[user]
            bossBar!!.setProgress(double)
        }
    }
}